### DISCUSSION TITLE
Cannot start a second lnd on my local machine in regtest

### DISCUSSION THREAD

**Discussion by Luca-Fischer** [lightningnetwork/lnd#8761] [Troubleshooting]
Hello, 

I am running a bitcoin node in regtest on macOS and try to run mulitple lnd nodes to test out the lightning network. 
Here is my bitcoin.conf:
`
rpcuser=bitcoinrpc
rpcpassword=OaKWn2TLfJBq-Lvjssq-...

zmqpubrawblock=tcp://127.0.0.1:28332
zmqpubrawtx=tcp://127.0.0.1:28333

estimatesmartfee=3

txindex=1

regtest=1

daemon=1
`

Bitcoin starts with bitcoind. 

Then I run my first lnd with lnd1. Here is the .bash_profile and lnd.conf:
`
export LND_DIR="$HOME/Library/Application\ Support/lnd"
export MACAROONPATH="$LND_DIR/data/chain/bitcoin/regtest/admin.macaroon"
      
alias lnd1="lnd --lnddir=$LND_DIR";
alias lncli1="lncli --lnddir=$LND_DIR --network regtest --macaroonpath $MACAROONPATH --tlscertpath $LND_DIR/tls.cert"

export LND_DIR_2="$HOME/Library/Application\ Support/lnd2"
export MACAROONPATH_2="$LND_DIR_2/data/chain/bitcoin/regtest/admin.macaroon"

alias lnd2="lnd --lnddir=$LND_DIR_2";
alias lncli2="lncli --lnddir=$LND_DIR_2 --network regtest --macaroonpath $MACAROONPATH_2 --tlscertpath $LND_DIR_2/tls.cert"
`

lnd.conf for lnd1:
`
[Application Options]
debuglevel=info
maxpendingchannels=5
alias=regtest-lnd
color=#123456
restlisten=0.0.0.0:8081
rpclisten=0.0.0.0:10009
tlsextraip=0.0.0.0

[Bitcoin]
bitcoin.active=1
bitcoin.regtest=1
bitcoin.node=bitcoind

[Bitcoind]
bitcoind.rpcuser=bitcoinrpc
bitcoind.rpcpass=OaKWn2TLfJBq-Lvjssq-...
bitcoind.zmqpubrawblock=tcp://127.0.0.1:28332
bitcoind.zmqpubrawtx=tcp://127.0.0.1:28333
`

Everything works, I can create a wallet with lncli1 create and so on. 
With the second lnd2 I have trouble.
My lnd.conf for lnd2:
`
[Application Options]
debuglevel=info
maxpendingchannels=5
alias=regtest-lnd-sender
color=#654321
restlisten=0.0.0.0:8083
rpclisten=0.0.0.0:10011
tlsextraip=0.0.0.0

[Bitcoin]
bitcoin.active=1
bitcoin.regtest=1
bitcoin.node=bitcoind

[Bitcoind]
bitcoind.rpcuser=bitcoinrpc
bitcoind.rpcpass=OaKWn2TLfJBq-Lvjssq-...
bitcoind.zmqpubrawblock=tcp://127.0.0.1:28334
bitcoind.zmqpubrawtx=tcp://127.0.0.1:28335
`

I can run the command lncli2 create but after entering a password, n, 'enter' I get: 

`
[lncli] unable to generate seed: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: tls: failed to verify certificate: x509: certificate signed by unknown authority (possibly because of \"x509: ECDSA verification failure\" while trying to verify candidate authority certificate \"MBP-von-MyName.fritz.box\")"
`

When I do not run lnd1 but run lnd2 and try to create a wallet I get a different error:

`
[lncli] unable to generate seed: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:10009: connect: connection refused"
`

When I test the network connection I get: 

`
nc -v localhost 10011
nc: connectx to localhost port 10011 (tcp) failed: Connection refused
Connection to localhost port 10011 [tcp/*] succeeded!
`

And for this:

`
nc -v localhost 10009
nc: connectx to localhost port 10009 (tcp) failed: Connection refused
nc: connectx to localhost port 10009 (tcp) failed: Connection refused
`

But the first lnd does start.

I am completely clueless. 
Does anyone can help?

**Comment by guggero** [comment]
> `x509: ECDSA verification failure...`

This means you're not specifying the correct TLS certificate to use in your `lncli2` command. Make sure you point `--lnddir` to the correct directory.

> `127.0.0.1:10009: connect: connection refused`

That means `lnd` isn't responding on that port. Because lnd2 is running on `10011` as you configured in the config. So whenever you're trying to talk to lnd2, use `lncli --rpcserver localhost:10011 --lnddir <see_comment_above>`.

Basically your problem is that when using `lncli` you correctly need to address each node by its data directory and port it is running on.

**Comment by Luca-Fischer** [comment]
Thanks!
